packages/grid/x-data-grid/src/material/icons/GridColumnUnsortedIcon.tsx

```diff
@@ -3,7 +3,7 @@ import * as React from 'react';
 import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
 import { GridSortDirection } from '../../models/gridSortModel';
 
-interface GridColumnUnsortedIconProps extends SvgIconProps {
+export interface GridColumnUnsortedIconProps extends SvgIconProps {
```

Comment:
I think exporting it this way is not gonna work because:
1. This component `GridColumnUnsortedIcon` is a slot and differently exists for [material](https://github.com/mui/mui-x/blob/e5d35580b78f5b8bb97ebf0cab2a3775f015418a/packages/grid/x-data-grid/src/material/icons/GridColumnUnsortedIcon.tsx#L6-L8) and [joy](https://github.com/mui/mui-x/blob/e5d35580b78f5b8bb97ebf0cab2a3775f015418a/packages/grid/x-data-grid/src/joy/icons.tsx#L289-L291) design systems.
2. Since this file doesn't export like `export * from './GridColumnUnsortedIcon'`, this added export will make no difference anyways.

**Possible Options:**
1. We document is somehow and users keep defining on their side.
2. We export it from the specific design entry point, so the users will import it like
   a. import { GridColumnUnsortedIconProps } from '@mui/x-data-grid/material';
   b. import { GridColumnUnsortedIconProps } from '@mui/x-data-grid/joy';
3. Export them from the root with different names, like `GridColumnUnsortedIconPropsMaterial`, `GridColumnUnsortedIconPropsJoy`.

To me 1 and 2 feel better, 3 I would not consider since this will nagate the initial idea of isolating the exports for different design systems.
For 2 though, I'm not sure if it'll be discoverable enough for the users, since it will not be obvious from what path they need to import this as we have all other imports from the root path.

Looping @cherniavskii since the problem is a bit broader, should apply to all other design system related exports. Would be nice to hear how you'd approach this.

CC @michelengelen

Reply:
Makes absolute sense @MBilalShafi ... maybe it would make sense to add a recipe to the docs instead? We should at least add a short section about this (and similar cases).
WDYT?